### ISSUE TITLE
Compile time validation of key bindings

### ISSUE THREAD

**Issue by sminez** [sminez/penrose#107]
### Is your feature request related to a problem? Please describe

Currently, user keybindings are parsed and validated at runtime. This means that it is possible for users to inadvertantly break their window manager by defining invalid keybindings. This is fixable but can lead to needing to edit files / compile from a TTY which is less than ideal.

### Describe the change / addition you'd like to see made

Simple validation of keybindings can be opted in for at compile time:
- Key names and modifiers are all valid
- Bindings have not been repeated.

### Additional context

As it stands, the `map` and `refmap` sections of the `gen_keybindings` macro would not be verifyable in a proc macro, due to taking an iterator for defining the source mapping being used in the bindings themselves. It might be worth deprecating this in favour of explicit literals like so:

```rust
// current
        refmap [ config.ws_range() ] in {
            "M-{}" => focus_workspace [ index_selectors(config.workspaces().len()) ];
            "M-S-{}" => client_to_workspace [ index_selectors(config.workspaces().len()) ];
        };
```
```rust
// proposed
        refmap [ 1, 2, 3, 4, 5, 6, 7, 8, 9 ] in {
            "M-{}" => focus_workspace [ index_selectors(config.workspaces().len()) ];
            "M-S-{}" => client_to_workspace [ index_selectors(config.workspaces().len()) ];
        };
```

To be honest, the current `map` and `refmap` sections have always felt hacky and could do with improving. For now, this keybinding validation should limit itself to validating the main section only.

**Comment by sminez** [comment]
The above proposed change to `refmap` and `map` is different from what was eventually settled on: see PR #110 for details.

**Comment by sminez** [comment]
implemented in #110
